pageserver/src/tenant/mgr.rs

```diff
@@ -151,61 +151,86 @@ async fn safe_rename_tenant_dir(path: impl AsRef<Utf8Path>) -> std::io::Result<U
 
 static TENANTS: Lazy<RwLock<TenantsMap>> = Lazy::new(|| RwLock::new(TenantsMap::Initializing));
 
-/// Initialize repositories with locally available timelines.
-/// Timelines that are only partially available locally (remote storage has more data than this pageserver)
-/// are scheduled for download and added to the tenant once download is completed.
-#[instrument(skip_all)]
-pub async fn init_tenant_mgr(
-    conf: &'static PageServerConf,
-    resources: TenantSharedResources,
-    init_order: InitializationOrder,
-    cancel: CancellationToken,
-) -> anyhow::Result<()> {
-    // Scan local filesystem for attached tenants
-    let tenants_dir = conf.tenants_path();
+fn emergency_generations(
```

Comment:
Again thinking about function ordering, since this is one trivial collect, wondering if it should be inlined instead, but this is me trying to see through the refactorings to understand what really changed.

Reply:
The idea with this one is just that it sits next to init_load_generations, and acts as a way of isolating all the emergency-specific logic in one function